envision/web/src/components/app.js

```diff
@@ -67,20 +67,20 @@ function App({ client }) {
   const matchedSimulationId = routeMatch ? routeMatch.params.simulation : null;
 
   useEffect(() => {
-    const fetchRunningSim = async () => {
+    (async () => {
       let ids = await client.fetchSimulationIds();
       if (ids.length > 0) {
         if (!matchedSimulationId || !ids.includes(matchedSimulationId)) {
           history.push(`/${ids[ids.length - 1]}`);
         }
       }
       setSimulationIds(ids);
-    };
+    })();
 
     // checks if there is new simulation running every 3 seconds.
-    const interval = setInterval(fetchRunningSim, 3000);
-    return () => clearInterval(interval);
-  }, [matchedSimulationId]);
+    // const interval = setInterval(fetchRunningSim, 3000);
+    // return () => clearInterval(interval);
```

Comment:
I had to comment out these two lines since they kept on reconnecting the server to a running Simulation when you switch to "All Simulation" or multi-instance display page. The downside now is that the user needs to reload the server every time a new Simulation is added to the server port.

Reply:
One thing we can do is to check if there is a change in the number of ids. If there is, we can call the `fetchRunningSim`, else we don't set the interval for `fetchRunningSim`.

Reply:
Ok , I tried this and I don't think this can work.

Reply:
UPDATE: This has been fixed now!